### PR TITLE
Added option to customize order of categories

### PR DESCRIPTION
**Pull request by alesdotio**
It's very common to want to be able to manually specify the order of your categories. For example, a "News" or "FAQ" category will have better visibility if it's on top.

I added an "order" field to Category and also added a test. This change is backwards compatible.

### CODE REVIEW

**Comment by henri-hulski** [comment]
:+1:

File: spirit/category/models.py [hunk 1]
```diff
@@ -21,13 +21,15 @@ class Category(models.Model):
     is_closed = models.BooleanField(_("closed"), default=False)
     is_removed = models.BooleanField(_("removed"), default=False)
     is_private = models.BooleanField(_("private"), default=False)
+    order = models.IntegerField(_('order'), default=10,
```
Comment (nitely) [inline spirit/category/models.py]: Is there a reason the default is 10?

**Comment by nitely** [comment]
@alesdotio features should be discussed before making pull requests. You can start a discussion on community.spirit. I'll probably create a gitter later.

**Comment by alesdotio** [comment]
Reason for the default being 10:
It is slightly more convenient if you want to change the order of only one category, for example, to sort a category to the top you would just set that one to 1. If the default was 0, then you would have to increase the order of *all* other categories but the one you want on top (or set it to -1, but that can get confusing).

**Comment by alesdotio** [comment]
> features should be discussed before making pull requests

Sorry, I was not aware of this. Is there specific instructions on contributing anywhere?

**Comment by nitely** [comment]
No, there is not, my bad. You usually open discuss it somewhere (ie: gitter or a mail list) and then you open an issue for future references. I believe it's a general rule for open source projects, though, I understand not everyone is aware of this.

I will write them in the readme ASAP.

**Comment by nitely** [comment]
> @alesdotio
> Reason for the default being 10:
> It is slightly more convenient if you want to change the order of only one category

I think if you have change several categories, and then want to add a new one after a specific category, you would have to go and change all the orders, right?

Some thoughts:
We could add a dropdown with the categories instead of the order number, and just reorder them, changing the number, when a category is edited/created. We would lose the order by title, though...

**Comment by alesdotio** [comment]
IMO, Ideally the UI would be drag and drop on the category list and the order field would not be shown at all. Setting numbers to order stuff is not the most intuitive and you run into problems eventually, as you said. If you agree, I can update this PR when I add the dragndrop.

**Comment by alesdotio** [comment]
Another option would be to use [django-mptt](http://django-mptt.github.io/django-mptt/) for Categories which would also come in handy if you plan to have deeper category nesting. It abstracts away the whole ordering and structure and is the fastest way to query trees.

**Comment by henri-hulski** [comment]
I like the drag 'n drop solution from the mezzanine admin (I think it's actually grapelli_save).
It's very lightweight and minimalistic and works over several levels in an intuive way.

**Comment by nitely** [comment]
> IMO, Ideally the UI would be drag and drop on the category list and the order field would not be shown at all. Setting numbers to order stuff is not the most intuitive and you run into problems eventually, as you said. If you agree, I can update this PR when I add the dragndrop.

I've yet to see a drag'n'drop library that works well on desktop and mobile and it is lightweight. IMHO, I'd just an up and down arrow that changes the order.

Please, notice all Spirit's JS code is (unit tested) coffeescript.

> Another option would be to use django-mptt for Categories which would also come in handy if you plan to have deeper category nesting. It abstracts away the whole ordering and structure and is the fastest way to query trees.

I've taken the decision of not having more than one level of nesting a while ago. I'm glad most new forums out there have done the same :smile:

**Comment by henri-hulski** [comment]
A nice drag and drop library is Dart Drag and Drop on https://github.com/marcojakob/dart-dnd. 
It supports mobiles and has options like verticalOnly and sortable which could be useful here. But it's written in dart, so I don't know if it's an option here.

**Comment by nitely** [comment]
It seems awesome, but according to my web console the js file size is about 142,2 kB :expressionless:

**Comment by henri-hulski** [comment]
You're right, it not seems to be lightweight at all.
I've found another modern minimalistic sortable library in pure javascript, only depending on html5:
http://rubaxa.github.io/Sortable
It works fine on mobile and desktop and has only 26 kB (11,4 kB minified).

**Comment by henri-hulski** [comment]
Alternatively we can use drag and drop on desktop and 2 little up and down arrows on mobile.

http://absolutelysuperlist.avocado.io demonstrates a very clean solution. Just replace `up/down` with the arrows and `remove` with an `x` icon, delete `Duplicate Move:` and you're done.

The library is an jquery plugin which supports also older browsers (13,8 kB / 5,9 kB min).

I think this is less invasive and maybe even better for mobiles, where dragging and scrolling can confuse each other.

**Pull request closed without merging**